dense_vector/track.py

```diff
@@ -0,0 +1,32 @@
+import random
+
+
+def knn_param_source(track, params, **kwargs):
+    # ensure the sequence of vector queries is the same from run to run
+    random.seed(31)
+    # choose a suitable index: if there is only one defined for this track
+    # choose that one, but let the user always override index
+    if len(track.indices) == 1:
+        default_index = track.indices[0].name
+    else:
+        default_index = "_all"
+
+    index_name = params.get("index", default_index)
+
+    return {
+        "body": {
+            "knn": {
+                "field": "vector",
+                "query_vector": params.get("query-vector", [random.uniform(-1.0, 1.0) for _ in range(96)]),
```

Comment:
Sorry @DJRickyB , I missed an important consideration here. The vector similarity we're using requires that all vectors have unit length. So we'll need to normalize these random vectors. Here's an example using NumPy, but it's a simple formula that we could also code by hand: https://stackoverflow.com/questions/21030391/how-to-normalize-a-numpy-array-to-a-unit-vector. Currently the searches will fail if we try to use a random query vector.

Alternatively, would it be possible to check in a "query vector" file of ~1000 vectors and use that to create all queries? We have such a file available, and the vectors are already normalized, and more realistic than random vectors.